src/main/java/org/cactoos/iterable/Mapped.java

```diff
@@ -51,7 +51,7 @@ public Mapped(final Func<X, Y> fnc, final X... src) {
      * @param fnc Func
      * @param src Source iterable
      */
-    public Mapped(final Func<X, Y> fnc, final Iterable<X> src) {
+    public Mapped(final Func<X, Y> fnc, final Iterable<? extends X> src) {
```

Comment:
@andreoss here I think we need:
- to move the the `X` argument at the constructor level, it's not needed at the class level IMHO
- have `Func<? super X, ? extends Y>` to be able to express more things

Reply:
@victornoel Added todo for removing `X` as a class level type parameter

Reply:
@victornoel Actually, I've done it in this PR.